File: backend/stock_control/src/main/java/com/ufba/stock_control/entities/TransactionType.java
package com.ufba.stock_control.entities;

import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "transaction_type")
public class TransactionType {
  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;
  
  @Column(name = "transaction_direction")
  private TransactionDirection direction;
  
  @Column(name = "description")
  private TransactionDescription description;
}

File: FrontEnd/sistema-controle-estoque/src/types/transaction_type.ts
import type { TRANSACTION_TYPE_DESCRIPTION } from '@/enums/transaction_type_description'
import type { TRANSACTION_TYPE_DIRECTION } from '@/enums/transaction_type_direction'

export type TransactionType = {
  id: string
  direction: TRANSACTION_TYPE_DIRECTION
  description: TRANSACTION_TYPE_DESCRIPTION
}
